### PR TITLE
[NUI] Minor refactoring CollectionView and RecyclerView

### PR DESCRIPTION
**Pull request by everLEEst**
Minor refactoring.
- Use as type casing.
- Renaming Function.
- Divide long functions to small functions.

### API Changes ###

 - ACR:

### CODE REVIEW

**Review by taehyub — APPROVED** [review]
LGTM

File: src/Tizen.NUI.Components/Controls/RecyclerView/CollectionView.cs [hunk 1]
```diff
@@ -673,7 +729,7 @@ public override void OnRelayout(Vector2 size, RelayoutContainer container)
             base.OnRelayout(size, container);
 
             wasRelayouted = true;
-            if (needInitalizeLayouter) Init();
+            if (needInitalizeLayouter) ReinitializeLayout();
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Controls/RecyclerView/CollectionView.cs:732]: do you want to change this as follows?
```
if (needInitalizeLayouter)
{
    ReinitializeLayout();
}
```

Comment (everLEEst) [reply src/Tizen.NUI.Components/Controls/RecyclerView/CollectionView.cs:732]: oh I missed it. thank you for noticing. will fix right now.

**Review by jaehyun0cho — APPROVED** [review]
LGTM

**Pull request merged**